### DISCUSSION TITLE
When would /tmp folder get cleared?

### DISCUSSION THREAD

**Discussion by AntonioSun** [deta/cloud-docs#581] [Help] [answered]
> You get a incoming request on your server, the micro awakes with an execution ram of 128mb ( 128mb solely for your program), after 10s it gets deleted. It happens for each request.

My micro write files to the `/tmp` folder for caching. I found it get cleared after a while, definitely not just 10 sec, but definitely get cleared after a while. 

So my question is when would /tmp folder get cleared? Is it really only 10sec, and what I get after 10sec is cached somewhere along the path from micro to my machine, or it'll be some later time for some other reason?

**Comment by lemonyte** [comment, marked as answer]
There is no guarantee as to when it gets cleared. You should assume that everything in the `/tmp` directory is deleted as soon as the instance is terminated after 10 seconds, and no data persists between instances. There might be some caching on Deta's end which makes it look like it persists between instances.

**Reply by AntonioSun** [reply]
I was so curious that I did a test, and my test (single one) shows that it get cleared after 25min.
